src/main/java/com/purbon/kafka/topology/roles/ResourceFilter.java

```diff
@@ -1,82 +1,80 @@
+  private final List<String> managedTopicPrefixes;
+  private final List<String> managedGroupPrefixes;
+
+  public ResourceFilter(Configuration config) {
+    this.managedServiceAccountPrefixes = config.getServiceAccountManagedPrefixes();
+    this.managedTopicPrefixes = config.getTopicManagedPrefixes();
+    this.managedGroupPrefixes = config.getGroupManagedPrefixes();
+  }
+
+  public boolean matchesManagedPrefixList(TopologyAclBinding topologyAclBinding) {
+    String resourceName = topologyAclBinding.getResourceName();
+    String principle = topologyAclBinding.getPrincipal();
+    // For global wild cards ACL's we manage only if we manage the service account/principle,
+    // regardless. Filtering by service account will always take precedence if defined
+    if (haveServiceAccountPrefixFilters() || resourceName.equals("*")) {
+      if (resourceName.equals("*")) {
+        return matchesServiceAccountPrefixList(principle);
+      } else {
+        return matchesServiceAccountPrefixList(principle)
+            && matchesTopicOrGroupPrefix(topologyAclBinding, resourceName);
+      }
+    } else if (haveTopicNamePrefixFilter() || haveGroupNamePrefixFilter()) {
+      return matchesTopicOrGroupPrefix(topologyAclBinding, resourceName);
     }
 
-    public boolean matchesManagedPrefixList(TopologyAclBinding topologyAclBinding) {
-        String resourceName = topologyAclBinding.getResourceName();
-        String principle = topologyAclBinding.getPrincipal();
-        // For global wild cards ACL's we manage only if we manage the service account/principle,
-        // regardless. Filtering by service account will always take precedence if defined
-        if (haveServiceAccountPrefixFilters() || resourceName.equals("*")) {
-            if (resourceName.equals("*")) {
-                return matchesServiceAccountPrefixList(principle);
-            } else {
-               return matchesServiceAccountPrefixList(principle)
-                       && matchesTopicOrGroupPrefix(topologyAclBinding, resourceName);
-            }
-        } else if (haveTopicNamePrefixFilter() || haveGroupNamePrefixFilter()) {
-            return matchesTopicOrGroupPrefix(topologyAclBinding, resourceName);
-        }
-
-        return true; // should include everything if not properly excluded earlier.
-    }
-
-    private boolean matchesTopicOrGroupPrefix(
-            TopologyAclBinding topologyAclBinding, String resourceName) {
-        if ("TOPIC".equalsIgnoreCase(topologyAclBinding.getResourceType())) {
-            return matchesTopicPrefixList(resourceName);
-        } else if ("GROUP".equalsIgnoreCase(topologyAclBinding.getResourceType())) {
-            return matchesGroupPrefixList(resourceName);
-        }
-        return false;
-    }
-
-    private boolean matchesTopicPrefixList(String topic) {
-        return matchesPrefix(managedTopicPrefixes, topic, "Topic");
-    }
-
-    private boolean matchesGroupPrefixList(String group) {
```

Comment:
sorry I have to mention, bloody formatter, I do really have to get a better flow to use this formatter tool ... sorry for the pain.

Reply:
`mvn fmt:check` does not complain :(
I will try to check why.

Reply:
@purbon I think we should bump the version of the plugin to the spotify 2.19 introducing the failOnError option (default is true).
But the current file seems to be correctly formatted already (it wasn't). Was it the meaning of your comment ? :p

Reply:
>  Was it the meaning of your comment?

No, no. It bothers me the formatter changes way too much code *all* the time. I would love to implement a way the PR's include minimal redundant changes in formatting, ideally zero only the code changes. Not sure I explain. 
This initial flow is my fault, might be laziness not to implement it better with all my other responsibilities. :-)